pytensor/graph/basic.py

```diff
@@ -1896,11 +1896,10 @@ def equal_computations(
             if isinstance(x, Constant):
                 return np.array_equal(x.data, y)
             return False
-        if x.owner and not y.owner:
+        xy_have_owner = (x.owner is None, y.owner is None)
+        if any(xy_have_owner) and not all(xy_have_owner):
             return False
-        if y.owner and not x.owner:
-            return False
-        if x.owner and y.owner:
+        if all(xy_have_owner):
```

Comment:
I find this a bit less readable now

Reply:
Ah and I even reversed it :upside_down_face: